### PR TITLE
remove tsfmt from repo

### PR DESCRIPTION
**Pull request by jikim-msft**
### Description

https://dev.azure.com/fluidframework/internal/_workitems/edit/2119/

This PR removes `typescript-formatter` package and `tsfmt` script across the repo. After removing the package from the repo, `prettier` will be enforcing the formatting rule

### CODE REVIEW

**Review by curtisman — APPROVED** [review]

File: packages/drivers/odsp-driver/src/test/localSnapshots/localSnapshot1.json [hunk 1]
```diff
@@ -140,9 +140,9 @@
...
....0.0\",\"@fluidframework/build-common\":\"^0.23.0\",\"@fluidframework/eslint-config-fluid\":\"^0.28.2000\",\"@fluidframework/test-tools\":\"^0.2.3074\",\"@rushstack/eslint-config\":\"^2.5.1\",\"@types/expect-puppeteer\":\"2.2.1\",\"@types/jest\":\"22.2.3\",\"@types/jest-environment-puppeteer\":\"2.2.0\",\"@types/node\":\"^14.18.0\",\"@types/puppeteer\":\"1.3.0\",\"@types/react\":\"^16.9.15\",\"concurrently\":\"^6.2.0\",\"cross-env\":\"^7.0.2\",\"css-loader\":\"^1.0.0\",\"eslint\":\"~8.6.0\",\"jest\":\"^26.6.3\",\"jest-junit\":\"^10.0.0\",\"jest-puppeteer\":\"^4.3.0\",\"less\":\"~3.9.0\",\"less-loader\":\"^4.1.0\",\"puppeteer\":\"^1.20.0\",\"rimraf\":\"^2.6.2\",\"style-loader\":\"^1.0.0\",\"ts-loader\":\"^9.3.0\",\"typescript\":\"~4.5.5\",\"typescript-formatter\":\"7.1.0\",\"url-loader\":\"^2.1.0\",\"webpack\":\"^5.72.0\",\"webpack-cli\":\"^4.9.2\",\"webpack-dev-server\":\"~4.6.0\",\"webpack-merge\":\"^5.8.0\"},\"fluid\":{\"browser\":{\"umd\":{\"files\":[\"dist/main.bundle.js\"],\"library\":\"main\"}}},\"jest-junit\":{\"outputDirectory\":\"nyc\",\"outputName\":\"jest-junit-report.xml\"}},\"config\":{}},\"approvalSequenceNumber\":0,\"commitSequenceNumber\":0,\"sequenceNumber\":0}]]",
```
Comment (Josmithr) [inline packages/drivers/odsp-driver/src/test/localSnapshots/localSnapshot1.json:143]: This change seems unrelated to the formatter dependency removal. Probably worth reverting (unless I'm missing something).

**Review by Josmithr — COMMENTED** [review]
These changes look good to me, aside from the one comment I left. That said, I would be interested in having either @tylerbutler or @curtisman review the changes to fluid-build to ensure there aren't any other pieces we need to change for this.

Edit: @curtisman beat me to it! 😂 Looks good to me then!

**Review by Josmithr — APPROVED** [review]

**Review by tylerbutler — APPROVED** [review]
I recommend removing the fluid-build changes from this PR and doing them separately. Easier to revert, and we may not want to remove tsfmt support in fluid-build _immediately_, for those who install globally.

**Comment by jikim-msft** [comment]
> I recommend removing the fluid-build changes from this PR and doing them separately. Easier to revert, and we may not want to remove tsfmt support in fluid-build _immediately_, for those who install globally.

@tylerbutler Just to confirm. Your suggestion is to leave the `tsfmt` command inside the `build-tools` for this PR and make a separate one just in case we need to revert it in the future?

File: build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/leafTask.ts [hunk 2]
```diff
@@ -47,7 +47,7 @@ export abstract class LeafTask extends Task {
     }
 
     public get isDisabled() {
-        const isLintTask = this.executable === "eslint" || this.executable === "tsfmt";
+        const isLintTask = this.executable === "eslint";
```
Comment (tylerbutler) [inline build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/leafTask.ts:50]: @jikim-msft These are the fluid-build changes I was talking about. This file and the related change in taskFactory.ts. If you don't revert these changes then the next release of fluid-build (which many folks use to build the repo) won't parse tsfmt tasks correctly. This may mean that fluid-build will stop working on older branches where the tsfmt task still exists. So I think it's safest to leave fluid-build able to parse tsfmt tasks for the time being.

**Review by tylerbutler — APPROVED** [review]
Since you're updating dependencies, be sure you run npm install and check in lock file changes.
